lziprecover: init at 1.22

vlaci opened the pull request:
Maintainer is absent for the original package that's why I haven't put myself in here. I can add myself to both of them if needed

###### Motivation for this change

This is a data recovery tool for `lzip` files. which I am depending on for my current job :)

###### Things done

Tried cross-building to multiple platform successfully but didn't test any of the crossbuilt output. CFLAGS and co. are directly lifted over from the `lzip` defivation.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

vlaci committed:
lzip: License is GPL-2 or later

According to the homepage and `COPYING` file inside the source
archive.

vlaci committed:
lziprecover: init at 1.22

Companion package to `lzip`

c0bw3b reviewed (commented):
It builds and runs properly.

One message during build though:

```session
testing --reproduce...
warning: skipping --reproduce test: lzip not found or not the right version.
Try 'make LZIP_NAME=<name_of_lzip_executable> check'.
```

You need to pass `lzip` in `checkInputs` in order to run all possible tests during check phase.

SuperSandro2000 commented on pkgs/tools/compression/lziprecover/default.nix:
```diff
@@ -0,0 +1,28 @@
+    homepage = "https://www.nongnu.org/lzip/lziprecover.html";
+    description = "Data recovery tool for lzip compressed files";
+    license = lib.licenses.gpl2Plus;
+    platforms = lib.platforms.all;
```
Missing maintainer.

SuperSandro2000 commented on pkgs/tools/compression/lziprecover/default.nix:
```diff
@@ -0,0 +1,28 @@
+    "CFLAGS=-O3"
+    "CXXFLAGS=-O3"
+  ] ++ lib.optional (stdenv.hostPlatform != stdenv.buildPlatform)
+    "CXX=${stdenv.cc.targetPrefix}c++";
```
Can we apply this uncondtional?

vlaci replied:
I guess we can, I modified lzip too, from which this idiom was copied.

vlaci committed:
lzip: simplify configureFlags

It is not an issue if $CXX variable is always set to an absolute
value, there is no need to set it conditionally.

vlaci committed:
lzip: nominate myself as maintainer

SuperSandro2000 commented:
We can either put this entire PR onto staging or only move the lzip things there because they cause to many rebuilds

SuperSandro2000 commented:
We can either change the entire PR to target staging or only move the lzip changes to staging.

vlaci commented:
I've retargeted the PR to the `staging` branch
